### PR TITLE
add LSTMGridRandomRecipe and MTNetGridRandomRecipe for zouwu

### PR DESCRIPTION
**Pull request by shane-huang**
This is to make it easier for users to config serach space in zouwu. Now they are able to replicate the performance tests shown in zouwu README with only a few argument changes in autots notebook instead of writing a new Recipe class. 

* add LSTMGridRandomRecipe and MTNetGridRandomRecipe for zouwu
* updated zouwu AutoTS notebook to use the new interface. 
* refactor automl recipe and add related unit tests
* some code style fixes and add api docs
* exclude zouwu/use-case from python wheel building

### CODE REVIEW

**Comment by yushan111** [comment]
LGTM

File: pyzoo/test/zoo/zouwu/test_auto_ts.py [hunk 1]
```diff
@@ -43,14 +43,35 @@ def create_data(self):
...
         tsp = AutoTSTrainer(dt_col="datetime",
                             target_col="value",
                             horizon=horizon,
                             extra_features_col=None
                             )
-        pipeline = tsp.fit(self.train_df, self.validation_df)
+        pipeline = tsp.fit(self.train_df,
+                           self.validation_df,
+                           recipe=LSTMGridRandomRecipe(
+                               num_rand_samples=1,
```
Comment (shane-huang) [inline pyzoo/test/zoo/zouwu/test_auto_ts.py:69]: num_rand_samples = 2

File: pyzoo/test/zoo/zouwu/test_auto_ts.py [hunk 2]
```diff
@@ -43,14 +43,35 @@ def create_data(self):
...
+                               batch_size=[1024],
+                               lstm_2_units=[8],
```
Comment (shane-huang) [inline pyzoo/test/zoo/zouwu/test_auto_ts.py:71]: grid search x 2

**Comment by jason-dai** [comment]
Please also update the user document

**Comment by shane-huang** [comment]
moved usage to github.io and make README.md more concise. @jason-dai

File: docs/mkdocs.yml [hunk 3]
```diff
@@ -138,6 +138,18 @@ pages:
 #    - Train, evaluate or predict a model: Keras2StyleAPIGuide/Optimization/training.md
 #    - Optimizers: Keras2StyleAPIGuide/Optimization/optimizer.md
 #    - Losses: Keras2StyleAPIGuide/Optimization/loss.md
+- Zouwu Solution:
```
Comment (jason-dai) [inline docs/mkdocs.yml:141]: Just `Zouwu`

File: docs/docs/Zouwu/overview.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+
+## Overview
+
```
Comment (jason-dai) [inline docs/docs/Zouwu/overview.md:3]: Need to have a title "Project Zouwu: Time Series Solution for Telco on Analytics Zoo"

File: docs/docs/Zouwu/overview.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+Project Zouwu provides a reference solution that is designed and optimized for common time series applications in the Telco industry, including:
+* _Use case_ - reference time series use cases in the Telco industry (such as network traffic forcasting, etc.)
+* _Model_ - built-in deep learning models for time series analysis (such as LSTM and [MTNet](https://arxiv.org/abs/1809.02105))
+* _AutoTS_ - AutoML support for building end-to-end time series analysis pipelines (including automatic feature generation, model selection and hyperparameter tuning).
+
+
+## Requirements
```
Comment (jason-dai) [inline docs/docs/Zouwu/overview.md:10]: why do we need requirements here?

File: docs/docs/Zouwu/overview.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+* requests
+* featuretools
+* pandas
+* Note that Keras is not needed to use Zouwu. But if you have Keras installed, make sure it is Keras 1.2.2. Other verisons might cause unexpected problems. 
+
+## Forecasting
+
+Time series forecasting has many applications in telco. Accurate forecast of telco KPIs (e.g. traffic, utilizations, user experience, etc.) for communication networks ( 2G/3G/4G/5G/wired) can help predict network failures, allocate resource, or save energy. Time series forecasting can also be used for log and metric analysis for data center IT operations for telco. Metrics to be analyzed can be hardware or VM utilizations, database metrics or servce quality indicators.
+
+We provided a reference use case where we forecast network traffic KPI's as a demo. Refer to [Network Traffic](./use-case/network-traffic) for forecasting.
```
Comment (jason-dai) [inline docs/docs/Zouwu/overview.md:29]: should refer to model & autots here

File: docs/docs/Zouwu/overview.md (hunk 5 above)
Comment (shane-huang) [reply docs/docs/Zouwu/overview.md:10]: Or maybe I can add it to tutorial

Comment (shane-huang) [reply docs/docs/Zouwu/overview.md:10]: it's needs to install dependencies besides analytics-zoo itself.

**Comment by shane-huang** [comment]
@jason-dai revised according to your comments.

**Comment by shane-huang** [comment]
PR validation has passed. merge for now and change docs later if needed.

**Pull request merged**